asset/filtered_manager.go

```diff
@@ -0,0 +1,65 @@
+package asset
+
+import (
+	"github.com/sensu/sensu-go/types"
+	"github.com/sensu/sensu-go/util/eval"
+)
+
+// NewFilteredManager returns an asset Getter that filters assets based on the
+// given entity. Assets that aren't filtered get passed to the underlying
+// getter, allowing composition with other asset managers.
+func NewFilteredManager(getter Getter, entity *types.Entity) *filteredManager {
+	return &filteredManager{
+		getter: getter,
+		entity: entity,
+	}
+}
+
+// filteredManager evaluates asset filter(s) and calls the underlying
+// asset Getter based on filter result.
+type filteredManager struct {
+	entity *types.Entity
```

Comment:
Rather than coupling Getter to entity, can we abstract this a bit and have a PredicatedGetter where the predicate is supplied and evaluated indirectly?

Reply:
You could then test your predicate separately from your predicated getter and your getter.

Reply:
I was so focused on `filteredManager` having a pluggable asset manager backend that I didn't even notice I could make the filter (predicate) pluggable as well. Good catch!